massa-graph/src/block_graph.rs

```diff
@@ -2678,6 +2675,46 @@ impl BlockGraph {
             })
     }
 
+    /// get the latest blockclique (or final) block ID at a given slot, if any
+    pub fn get_latest_blockclique_block_at_slot(&self, slot: &Slot) -> BlockId {
+        let (mut latest_final_block_id, mut latest_final_block_period) = self
+            .latest_final_blocks_periods
+            .get(slot.thread as usize)
+            .unwrap_or_else(|| panic!("unexpected not found latest final block period"));
+
+        self.max_cliques
+            .iter()
+            .filter(|c| c.is_blockclique)
+            .for_each(|clique| {
+                clique
+                    .block_ids
+                    .iter()
+                    .for_each(|id| match self.block_statuses.get(id) {
+                        Some(BlockStatus::Active {
+                            a_block,
+                            storage: _,
+                        }) => {
+                            if a_block.is_final {
+                                panic!(
+                                    "unexpected final block on getting latest blockclique at slot"
+                                );
+                            }
+                            if a_block.slot.thread == slot.thread
+                                && a_block.slot.period <= slot.period
+                                && a_block.slot.period > latest_final_block_period
+                            {
+                                latest_final_block_period = a_block.slot.period;
```

Comment:
the naming "latest_final_block_period / latest_final_block_id" is kinda wrong because it will almost never be a final block, I would recommend best_block_period / best_block_id